entity-service/src/test/kotlin/com/egm/stellio/entity/repository/Neo4jRepositoryTests.kt

```diff
@@ -464,31 +464,111 @@ class Neo4jRepositoryTests {
     }
 
     @Test
-    fun `it should delete an entity property`() {
+    fun `it should delete all property instances`() {
         val entity = createEntity(
             "urn:ngsi-ld:Beekeeper:1233",
             listOf("Beekeeper"),
             mutableListOf(
                 Property(name = "firstName", value = "Scalpa"),
+                Property(name = "firstName", value = "Scalpa2", datasetId = URI.create("urn:ngsi-ld:Dataset:firstName:2")),
                 Property(name = "lastName", value = "Charity")
             )
         )
-
-        neo4jRepository.deleteEntityProperty(entity.id, "lastName")
+        neo4jRepository.deleteEntityProperty(EntitySubjectNode(entity.id), "firstName", null, true)
 
         assertEquals(entityRepository.findById(entity.id).get().properties.size, 1)
         neo4jRepository.deleteEntity(entity.id)
     }
 
     @Test
-    fun `it should delete an entity relationship`() {
+    fun `it should delete the property default instance`() {
+        val entity = createEntity(
+            "urn:ngsi-ld:Beekeeper:1233",
+            listOf("Beekeeper"),
+            mutableListOf(
+                Property(name = "firstName", value = "Scalpa"),
+                Property(name = "firstName", value = "Scalpa2", datasetId = URI.create("urn:ngsi-ld:Dataset:firstName:2")),
+                Property(name = "lastName", value = "Charity")
+            )
+        )
+
+        neo4jRepository.deleteEntityProperty(EntitySubjectNode(entity.id), "firstName")
+
+        assertNull(entityRepository.findById(entity.id).get().properties.find {
```

Comment:
Arf.

Reply:
Also check it has not deleted the one with a datasetId

Reply:
> Arf.

Sorry i didn't understand what you mean

Reply:
> > Arf.
> 
> Sorry i didn't understand what you mean

It is a very complex expression, hard to read and follow.

Reply:
Any ideas about a simpler expression ?

Reply:
At least, split it. For instance, create an intermediate variable containing the properties, and then do your check on the properties. As a bonus, you'll have one less `findById(...)`.